### PR TITLE
Update to iD v2.17.0

### PR DESCRIPTION
**Pull request by quincylvania**
[Preview this release.](https://preview.ideditor.com/release) (Deployment may be delayed, check version number in bottom right)

# 2.17.0
#### :newspaper: News
* We've launched the [iD Blog](https://ideditor.blog) providing news and insights into the project from its maintainers and contributors ([#7045])

[#7045]: https://github.com/openstreetmap/iD/issues/7045

#### :mega: Release Highlights
* :metro: You can now add bridges and tunnels with a single click to fix crossing roads, rails, and waterways. Thanks to [@CarycaKatarzyna] for working on this!<br/>
_Find the "Add a bridge" and "Add a tunnel" fixes for each crossing in the Issues inspector._
* :earth_africa: Selected features now stay visible while zoomed out, plus you can zoom to multiple features together.<br/>
_Select a few large features and press <kbd>Z</kbd> to view their full extent, no matter how vast._
* :handshake: iD now has its own [Privacy Policy](https://github.com/openstreetmap/iD/blob/master/PRIVACY.md).<br/>
_Press <kbd>P</kbd> to view privacy preferences._

#### :boom: Breaking Changes
* Remove Maxar imagery layers due to [announced suspension of service](https://www.openstreetmap.org/user/@kevin_bullock/diary/391652)

#### :tada: New Features
* Display selected features at any zoom level ([#2962], [#5001])
* Add Privacy Policy and ability to opt-out of icons loaded from third-party sites ([#7040])

[#2962]: https://github.com/openstreetmap/iD/issues/2962
[#5001]: https://github.com/openstreetmap/iD/issues/5001
[#7040]: https://github.com/openstreetmap/iD/issues/7040

#### :sparkles: Usability
* Support squaring multiple selected features at the same time ([#6565])
* Support zooming to multiple selected features together with the <kbd>Z</kbd> shortcut ([#6696])
* Highlight the members of selected relations in yellow ([#5766])
* Return feature search results from all downloaded data, not just the visible area ([#6515])
* Show results for all three OpenStreetMap entity types when searching an ID without a prefix ([#7112])
* Style hotkeys in tooltips as keyboard keys ([#6574])
* Make the top toolbar horizontally-scrollable at narrow sizes ([#6755])
* Always show the Layer subfield of the Structure field when a value is present ([#6911])
* Disable the Circularize operation if the selected way is already circular ([#6816], [#6993], thanks [@CarycaKatarzyna])
* Fallback to a preset's vector icon if its image fails to load, e.g. due to content blockers ([#7028])
* Convert single-member multipolygons to basic areas when merging member lines ([#5085])
* Always show the currently selected background in the sources list ([#7061])
* Only show background sources with global coverage at low zooms ([#7062])
* Render features with a status-prefix tags with a dashed style, e.g. `demolished:building=yes`
* Add tooltips to Mapillary Map Features overlay icons ([#7079])
* Add button to manually retry connecting to the OpenStreetMap API upon a failure ([#6650])
* Clarify the OpenStreetMap API connection failure message ([#7021])
* Improve styling of points linked to Wikidata
* Render `landuse=village_green` areas in green ([#7011])
* Render Putting Greens and similar features in light green ([#7101])

[#6565]: https://github.com/openstreetmap/iD/issues/6565
[#6696]: https://github.com/openstreetmap/iD/issues/6696
[#5766]: https://github.com/openstreetmap/iD/issues/5766
[#6515]: https://github.com/openstreetmap/iD/issues/6515
[#7112]: https://github.com/openstreetmap/iD/issues/7112
[#6574]: https://github.com/openstreetmap/iD/issues/6574
[#6755]: https://github.com/openstreetmap/iD/issues/6755
[#6911]: https://github.com/openstreetmap/iD/issues/6911
[#6816]: https://github.com/openstreetmap/iD/issues/6816
[#6993]: https://github.com/openstreetmap/iD/issues/6993
[#7028]: https://github.com/openstreetmap/iD/issues/7028
[#5085]: https://github.com/openstreetmap/iD/issues/5085
[#7061]: https://github.com/openstreetmap/iD/issues/7061
[#7062]: https://github.com/openstreetmap/iD/issues/7062
[#7079]: https://github.com/openstreetmap/iD/issues/7079
[#6650]: https://github.com/openstreetmap/iD/issues/6650
[#7021]: https://github.com/openstreetmap/iD/issues/7021
[#7011]: https://github.com/openstreetmap/iD/issues/7011
[#7101]: https://github.com/openstreetmap/iD/issues/7101

[@CarycaKatarzyna]: https://github.com/CarycaKatarzyna

#### :white_check_mark: Validation
* For crossing ways issues, offer one-click "Add a bridge" and "Add a tunnel" fixes ([#6617], [#7055], thanks [@CarycaKatarzyna])
* For crossing way-building issues, offer one-click fixes that set higher or lower layers ([#5924], [#6911])
* Flag unclosed multipolgon parts ([#2223])
* Flag crossing and one-way issues for features with `waterway=fish_pass`
* Don't suggest upgrading to brands that don't exist in the feature's country ([#6513], [#6479])
* Don't flag very close points with differing house or unit numbers ([#6998])
* Allow the `not:brand:wikidata` tag to silence nonstandard brand warnings ([#6577])
* Include default field values when upgrading to a preset with a specific replacement ([#7033])
* Add tooltips to some disabled fix buttons
* Don't flag `natural=cape` or `amenity=vending_machine` on vertices as mismatched geometry ([#6982], [#6515])
* Don't add `oneway=yes` to `highway=motorway_link` by default ([#7013])
* Don't expect an arbitrary `junction` tag to imply a feature should be an area ([#6933])
* Prefer `aerialway=station` instead of `aerialway=yes` for aerialway stations ([#6994])
* Remove deprecation of `crossing=zebra` ([#6962])
* Remove deprecation of `amenity=social_club` and `leisure=social_club` ([#6252])
* Deprecate `agrarian=agrcultural_machinry` misspelling ([#7053])
* Deprecate `company=consulting`, `office=consultancy`, `office=consultant`, `shop=consulting`
* Deprecate `type=audio`, `type=video`, `type=caldera`, `type=extinct`, `type=scoria`, `type=shield`, `type=strato`, `type=extinct`
* Deprecate `amenity=research_institution`, `barrier=railing`, `craft=glass`, `man_made=gas_well`, `man_made=oil_well`, `man_made=village_pump`, `power=marker`

[#6617]: https://github.com/openstreetmap/iD/issues/6617
[#7055]: https://github.com/openstreetmap/iD/issues/7055
[#5924]: https://github.com/openstreetmap/iD/issues/5924
[#6911]: https://github.com/openstreetmap/iD/issues/6911
[#2223]: https://github.com/openstreetmap/iD/issues/2223
[#6513]: https://github.com/openstreetmap/iD/issues/6513
[#6479]: https://github.com/openstreetmap/iD/issues/6479
[#6998]: https://github.com/openstreetmap/iD/issues/6998
[#6577]: https://github.com/openstreetmap/iD/issues/6577
[#7033]: https://github.com/openstreetmap/iD/issues/7033
[#6982]: https://github.com/openstreetmap/iD/issues/6982
[#6515]: https://github.com/openstreetmap/iD/issues/6515
[#7013]: https://github.com/openstreetmap/iD/issues/7013
[#6933]: https://github.com/openstreetmap/iD/issues/6933
[#6994]: https://github.com/openstreetmap/iD/issues/6994
[#6962]: https://github.com/openstreetmap/iD/issues/6962
[#6252]: https://github.com/openstreetmap/iD/issues/6252
[#7053]: https://github.com/openstreetmap/iD/issues/7053

[@CarycaKatarzyna]: https://github.com/CarycaKatarzyna

#### :bug: Bugfixes
* Fix issue with rotating multiple points together ([#6977], [#6979], thanks [@hackily])
* Fix various instances where issue fixes might not get properly updated ([#6588], [#7037])
* Fix unexpected label offsets in Firefox 70 ([#7044])
* Don't move `area=yes` to nodes when using the Extract operation on areas ([#7057])
* Fix issue where the fills of unclosed multipolygon parts would not render entirely ([#2945])
* Prevent background tiles from appearing larger than expected ([#7070])
* Fix issue where additional fields would disappear immediately upon clearing their value ([#6580])
* Fix issue where adding a raw tag after deleting several would insert the blank row at the wrong index ([#7087])
* Fix issue where OpenStreetMap API error message would persist despite the download of new data ([#6650])
* Replace use of unsupported CSS property flagged by the OpenStreetMap website ([#7091])
* Fix issue where the selected Mapillary detection outline would not render in some browsers ([#6804])
* Fix issue where boundary relation members would not render correctly if they were also multipolygons members ([#6787])
* Fix issue where the Administrative Boundary preset was not properly overriding the Boundary preset ([#7118])
* Fix regression where the relation suggestion list could overflow the inspector ([#7115])
* Fix issue where the Unsquare Corners degree input could be too narrow in some browsers ([#7126], thanks [@iriman])
* Correct vertical centering of checkmark fields

[#6977]: https://github.com/openstreetmap/iD/issues/6977
[#6979]: https://github.com/openstreetmap/iD/issues/6979
[#6588]: https://github.com/openstreetmap/iD/issues/6588
[#7037]: https://github.com/openstreetmap/iD/issues/7037
[#7044]: https://github.com/openstreetmap/iD/issues/7044
[#7057]: https://github.com/openstreetmap/iD/issues/7057
[#2945]: https://github.com/openstreetmap/iD/issues/2945
[#7070]: https://github.com/openstreetmap/iD/issues/7070
[#6580]: https://github.com/openstreetmap/iD/issues/6580
[#7087]: https://github.com/openstreetmap/iD/issues/7087
[#6650]: https://github.com/openstreetmap/iD/issues/6650
[#7091]: https://github.com/openstreetmap/iD/issues/7091
[#6804]: https://github.com/openstreetmap/iD/issues/6804
[#6787]: https://github.com/openstreetmap/iD/issues/6787
[#7118]: https://github.com/openstreetmap/iD/issues/7118
[#7115]: https://github.com/openstreetmap/iD/issues/7115
[#7126]: https://github.com/openstreetmap/iD/issues/7126

[@hackily]: https://github.com/hackily
[@iriman]: https://github.com/iriman

#### :earth_asia: Localization
* Differentiate the "wireframe mode" and "highlight changes" hotkeys in the German localization ([#6972], thanks [@manfredbrandl])
* Improve Chinese address field ([#7075], thanks [@koaber])
* Add Bolivia-specific address and phone number formats ([#7147], thanks [@51114u9])
* Add the Occitan language to the Multilingual Name field ([#7156])

[#6972]: https://github.com/openstreetmap/iD/issues/6972
[#7075]: https://github.com/openstreetmap/iD/issues/7075
[#7147]: https://github.com/openstreetmap/iD/issues/7147
[#7156]: https://github.com/openstreetmap/iD/issues/7156

[@manfredbrandl]: https://github.com/manfredbrandl
[@koaber]: https://github.com/koaber
[@51114u9]: https://github.com/51114u9

#### :hourglass: Performance
* Determine locations' country codes without calling out to a geocoding server ([#6941])
* Reduce rendering lag considerably when many features are selected at once ([#3571])

[#6941]: https://github.com/openstreetmap/iD/issues/6941
[#3571]: https://github.com/openstreetmap/iD/issues/3571

#### :mortar_board: Walkthrough / Help
* Update links in the README to avoid http-to-https redirects ([#6984], thanks [@mbrickn])
* Add the <kbd>I</kbd> hotkey to the Keyboard Shortcuts list ([#6997])

[#6984]: https://github.com/openstreetmap/iD/issues/6984
[#6997]: https://github.com/openstreetmap/iD/issues/6997

[@mbrickn]: https://github.com/mbrickn

#### :rocket: Presets
* Add Notice Board, Poster Box, and Advertising Totem presets ([#6965], thanks [@hikemaniac])
* Add Kiddie Ride, Log Flume, and Swing Carousel presets ([#7039], thanks [@hikemaniac])
* Add Spice Shop preset ([#7031], thanks [@scaidermern])
* Add Giant Chess Board preset ([#7059], thanks [@ToastHawaii])
* Add Marker, Utility Marker, and Power Marker presets ([#6978])
* Add Access Aisle preset and style ([#7083])
* Add Research Institute preset and style ([#7078])
* Add Advanced Stop Line preset ([#7014])
* Add Lane Connectivity relation preset ([#7105])
* Add Water Tap preset ([#7066])
* Add Rail Yard preset ([#7119])
* Add unsearchable Disused Railway Feature preset ([#7119])
* Add Recently Demolished Building preset and render them as areas ([#7098])
* Add Recreational Network Node preset for Belgium, Germany, Luxembourg, and The Netherlands ([#6992])
* Add presets for new brands: Consultancy Office, Cleaning Service, Camera Equipment Store, Flooring Supply Shop, Pottery Store, Tool Rental
* Add Fish Pass preset
* Append "Area" to the names of linear area presets: Bridge, Tunnel, Road, River, Stream, Canal ([#7015])
* Append "Feature" to the names of various generic presets, e.g. "Tourism Feature"
* Append "Ride" to the names of some attraction presets, e.g "Pirate Ship Ride"
* Rename "Wood" preset to "Natural Wood"
* Rename "Car Pooling" and "Car Sharing" presets to "Car Pooling Station" and "Car Sharing Station"
* Rename "Pottery" craft preset to "Pottery Maker"
* Correct "Firepit" preset name to "Fire Pit"
* Correct capitalization of "J-Bar Lift" and "T-Bar Lift" preset names
* Update icons for Mast, Communication Mast, and Communication Tower ([#6985])
* Update icons for Gate, Kissing Gate, and Cattle Grid ([#6814], [#6489])
* Update icon for Park to be different from Tree ([#6633])
* Update icons for Bunker and Military Bunker ([#7139])
* Update icons for presets: diplomatic offices, marked crossings, transit platforms, buoys, Billboard, Jet Bridge, Scrap Yard, Bicycle Parking Garage, Bicycle Lockers, Bicycle Rental, Bicycle Repair Tool Stand, Boat Rental, Car Pooling Station, Car Sharing Station, Parking Lot, Multilevel Parking Garage, Underground Parking, Park & Ride Lot, Lean-To, Picnic Shelter, Transit Shelter, Block, Chain, Height Restrictor, Turnstile, Barn, Stable, Basket Maker, Boar Builder, Handicraft, Pottery, Indoor Corridor, Cycle & Foot Path, Street Lamp, Commemorative Plaque, Fire Pit, Pier, Floating Pier, Minaret, Tunnel Area, Water Tower, Grassland, Grass, Tree Row, Energy Supplier Office, Insurance Office, Slide, Water Slide, Play Structure, Underground Power Cable, Chocolate Store, Lighting Store, Motorcycle Repair Shop, Storage Rental, Art Installation, Sculpture, Statue, Coastline, Boat Store, Boatyard, Cabin, Holiday Cottage, Alpine Hut, Wilderness Hut, Hostel, Blacksmith
* Add "tree" as a search term for Natural Wood and Managed Forest ([#7097])
* Add "packstation" as a search term for package pickup and dropoff lockers ([#7052])
* Add "pilates" as a search term for Gym / Fitness Center ([#7137])
* Support limiting fields to specific countries ([#7085])
* Add GNIS Feature ID field to various preset for the United States ([#7086])
* Add VAT ID Number field to business presets for countries where VAT numbers are issued ([#6880])
* Add Wikimedia Commons Page field with link to view the page
* Add Mapillary ID field with link to view the image on the Mapillary website ([#7064])
* Add Internet Access, SMS, and Video Calls fields to the Telephone preset ([#7010])
* Add Tactile Paving field to the Steps preset ([#7082], thanks [@stragu])
* Add Reference Code field to Vending Machine presets ([#7002])
* Add Drinks field to the Drink Vending Machine preset
* Add Drinkable field to various water source presets
* Add Type field to Fountain preset
* Add Pump field to Water Well preset
* Add Utilities field to Utility Pole and Street Cabinet presets
* Add Brand field to more presets that could have brand tags
* Rename "Network Type" field for `network` to "Network Class"
* Add Network Type field for `network:type` to Route presets with a `network` value
* Rename "Suggested Hashtags" changeset field to just "Hashtags"
* Only show the Country field on Flagpole features with `flag:type=national` ([#7099])
* Don't show the Denomination field on features with `religion=none` ([#7135])

[#7139]: https://github.com/openstreetmap/iD/issues/7139
[#6965]: https://github.com/openstreetmap/iD/issues/6965
[#7039]: https://github.com/openstreetmap/iD/issues/7039
[#7031]: https://github.com/openstreetmap/iD/issues/7031
[#7059]: https://github.com/openstreetmap/iD/issues/7059
[#6978]: https://github.com/openstreetmap/iD/issues/6978
[#7083]: https://github.com/openstreetmap/iD/issues/7083
[#7078]: https://github.com/openstreetmap/iD/issues/7078
[#7014]: https://github.com/openstreetmap/iD/issues/7014
[#7105]: https://github.com/openstreetmap/iD/issues/7105
[#7066]: https://github.com/openstreetmap/iD/issues/7066
[#7119]: https://github.com/openstreetmap/iD/issues/7119
[#7098]: https://github.com/openstreetmap/iD/issues/7098
[#6992]: https://github.com/openstreetmap/iD/issues/6992
[#7015]: https://github.com/openstreetmap/iD/issues/7015
[#6985]: https://github.com/openstreetmap/iD/issues/6985
[#6814]: https://github.com/openstreetmap/iD/issues/6814
[#6489]: https://github.com/openstreetmap/iD/issues/6489
[#6633]: https://github.com/openstreetmap/iD/issues/6633
[#7097]: https://github.com/openstreetmap/iD/issues/7097
[#7052]: https://github.com/openstreetmap/iD/issues/7052
[#7137]: https://github.com/openstreetmap/iD/issues/7137
[#7085]: https://github.com/openstreetmap/iD/issues/7085
[#7086]: https://github.com/openstreetmap/iD/issues/7086
[#6880]: https://github.com/openstreetmap/iD/issues/6880
[#7064]: https://github.com/openstreetmap/iD/issues/7064
[#7010]: https://github.com/openstreetmap/iD/issues/7010
[#7082]: https://github.com/openstreetmap/iD/issues/7082
[#7002]: https://github.com/openstreetmap/iD/issues/7002
[#7099]: https://github.com/openstreetmap/iD/issues/7099
[#7135]: https://github.com/openstreetmap/iD/issues/7135

[@hikemaniac]: https://github.com/hikemaniac
[@scaidermern]: https://github.com/scaidermern
[@ToastHawaii]: https://github.com/ToastHawaii
[@stragu]: https://github.com/stragu

### CODE REVIEW

File: vendor/assets/iD/iD.css.erb [hunk 1]
```diff
@@ -774,6 +789,10 @@ path.fill.tag-power-plant {
 path.stroke.tag-natural-wetland {
     stroke: rgb(153, 225, 170);
 }
+path.fill.tag-natural-wetland {
+    stroke: rgb(153, 225, 170, 0.3);
+    fill: rgb(153, 225, 170, 0.3);
```
Comment (tomhughes) [inline vendor/assets/iD/iD.css.erb]: These two lines are causing a test failure - should they be `rgba` instead?

Comment (quincylvania) [reply vendor/assets/iD/iD.css.erb]: Thanks, I updated those lines.

File: vendor/assets/iD/iD.css.erb [hunk 2]
```diff
@@ -788,29 +807,41 @@ path.stroke.tag-natural-wetland {
...
     stroke: rgb(191, 232, 63);
 }
+path.fill.tag-landuse-cemetery,
+path.fill.tag-landuse-farmland,
+path.fill.tag-landuse-grass.tag-golf-green,
+path.fill.tag-landuse-meadow,
+path.fill.tag-landuse-orchard,
+path.fill.tag-landuse-vineyard {
+    stroke: rgb(191, 232, 63, 0.3);
+    fill: rgb(191, 232, 63, 0.3);
```
Comment (mmd-osm) [inline vendor/assets/iD/iD.css.erb]: Those two lines also need to be changed from rgb to rgba

Comment (mmd-osm) [reply vendor/assets/iD/iD.css.erb]: Hmm. this is already fixed in https://github.com/openstreetmap/iD/commit/acfd1af5b236a876bb9eb4e9b1fd6e402645c347, but I don't see, to find the fix for natural-wetland upstream (https://github.com/openstreetmap/iD/blob/master/css/25_areas.css#L197-L198). Somehow those two fixes are now a bit mixed up between repos. 🤷‍♂️

Comment (quincylvania) [reply vendor/assets/iD/iD.css.erb]: @mmd-osm Okay thanks, I just fixed this on that branch too.

Comment (mmd-osm) [reply vendor/assets/iD/iD.css.erb]: Ok thanks! Will you create another PR for this repo that fixes vineyards?

Comment (quincylvania) [reply vendor/assets/iD/iD.css.erb]: Ohhhh I wasn't reading carefully and didn't realize there were two separate issues here 😂

Will fix now.

**Pull request merged**